tests/utils/test_auth.py

```diff
@@ -42,3 +43,30 @@ def test_pypirc(fs):
     )
     assert credentials.username == 'guido'
     assert credentials.password == 'gat'
+
+
+def test_keyring_credentials(monkeypatch):
+    class MockKeyring:
+        @staticmethod
+        def get_credential(*_):
+            class Credential:
+                username = 'gat'
+
+            return Credential()
+
+        @staticmethod
+        def get_password(*_):
+            return 'guido'
```

Comment:
i have no strong opinion on this, but it would imo make sense to

a) realize this as a test fixture that is automatically used on all tests in this module.
b) add the `password` attribute to the `Credential` dummy.

Reply:
Thanks for the review. Good idea :)